src/pages/task/Breadcrumbs/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { useBreadcrumbAnalytics } from "analytics";
+import Breadcrumbs, { Breadcrumb } from "components/Breadcrumbs";
+import { getVersionRoute } from "constants/routes";
+import { useBreadcrumbRoot } from "hooks";
+import { shortenGithash } from "utils/string";
+
+interface TaskPageBreadcrumbsProps {
+  taskName: string;
+  patchNumber?: number;
+  versionMetadata?: {
+    id: string;
+    revision: string;
+    project: string;
+    isPatch: boolean;
+    author: string;
+    projectIdentifier: string;
+    message: string;
+  };
+}
+const TaskPageBreadcrumbs: React.VFC<TaskPageBreadcrumbsProps> = ({
+  versionMetadata,
+  patchNumber,
+}) => {
+  const { isPatch, author, projectIdentifier, id, message, revision } =
+    versionMetadata ?? {};
+  const breadcrumbRoot = useBreadcrumbRoot(isPatch, author, projectIdentifier);
+  const breadcrumbAnalytics = useBreadcrumbAnalytics();
+
+  const messageBreadcrumb = {
+    to: getVersionRoute(id),
+    text: message,
+    onClick: () => {
+      breadcrumbAnalytics.sendEvent({
+        name: "Click Link",
+        link: "version",
+      });
+    },
+    "data-cy": "bc-message",
+  };
+  const patchBreadcrumb = {
+    text: `Patch ${patchNumber}`,
+    "data-cy": "bc-patch",
+  };
+
+  const commitBreadcrumb = {
+    text: shortenGithash(revision),
+    "data-cy": "bc-version",
+  };
+
+  const breadcrumbs: Breadcrumb[] = [
+    breadcrumbRoot,
+    messageBreadcrumb,
+    isPatch ? patchBreadcrumb : commitBreadcrumb,
```

Comment:
yeah its a little confusing tbh i'm not sure if we should just omit the patchBreadcrumb or not it does technically contain different info since it shows the patch number or commit hash which isn't otherwise visible.

Reply:
could you combine the `messageBreadcrumb` + `patchBreadcrumb` into one thing?